pyScienceMode2/rehastim_interface.py

```diff
@@ -371,3 +367,527 @@ def get_motomed_torque(self) -> float:
+        connector_idx = (no_channel - 1) // 4
+        connector = connectors[connector_idx]
+
+        # Determine the channel
+        channel = channels[(no_channel - 1) % 4]
+
+        return channel, connector
+
+    #  Low level commands
+
+    def ll_init(self):
+        """
+        Initialize the lower level of the device. The low-level is used for defining a custom shaped pulse.
+        Each stimulation pulse needs to triggered from the computer.
+        """
+        ll_init = sciencemode.ffi.new("Smpt_ll_init*")
+        ll_init.high_voltage_level = (
+            sciencemode.lib.Smpt_High_Voltage_Default
+        )  # This switches on the high voltage source
+        ll_init.packet_number = self.get_next_packet_number()
+
+        if not sciencemode.lib.smpt_send_ll_init(self.device, ll_init):
+            raise RuntimeError("Low level initialization failed.")
+        self.log(
+            "Low level initialized",
+            "Command sent to rehastim: {}".format(self.TypeRehap24(sciencemode.lib.Smpt_Cmd_Ll_Init).name),
+        )
+
+        self.get_next_packet_number()
+        self._get_last_ack()
+        self.check_ll_init_ack()
+
+    def check_ll_init_ack(self):
+        """
+        Check the low level initialization status.
+        """
+        if not sciencemode.lib.smpt_get_ll_init_ack(self.device, self.ll_init_ack):
+            raise RuntimeError("Low level initialization failed.")
+        generic_error_check(self.ll_init_ack, self.ERROR_MAP)
+
+    def start_ll_channel_config(
+        self,
+        no_channel,
+        points=None,
+        stim_sequence: int = None,
+        pulse_interval: int | float = None,
+        safety: bool = True,
+    ):
+        """
+        Starts the low level mode stimulation.
+
+        Parameters
+        ----------
+        no_channel : int
+            The channel number [1,8].
+        points : list
+            Points to stimulate. [1,16]
+        stim_sequence : int
+            Number of stimulation sequence to be repeated.
+        pulse_interval : int | float
+            Interval between each stimulation sequence in ms.
+        safety : bool
+            Set to True if you want to check the pulse symmetry. False otherwise.
+        """
+        self._current_no_channel = no_channel
+        self._current_stim_sequence = stim_sequence
+        self._current_pulse_interval = pulse_interval
+        self.log("Low level stimulation started")
+
+        positive_area = 0
+        negative_area = 0
+
+        if points is None or len(points) == 0:
+            raise ValueError("Please provide at least one point for stimulation.")
+        channel, connector = self.channel_number_to_channel_connector(no_channel)
+        ll_config = sciencemode.ffi.new("Smpt_ll_channel_config*")
+
+        ll_config.enable_stimulation = True
+        ll_config.channel = channel
+        ll_config.connector = connector
+        ll_config.number_of_points = len(points)
```

Comment:
Add check for points 
If not isinstance(points, list):
    points = [points]
for point in points: 
if not insinstance(point, Point):
     raise Runtime error("point must be of type Point.)

Reply:
Done.